core/contracts/financial-templates/implementation/Liquidatable.sol

```diff
@@ -171,26 +171,49 @@ contract Liquidatable is PricelessPositionManager {
      * @dev This method generates an ID that will uniquely identify liquidation
      * for the sponsor.
      * @param sponsor address to liquidate
-     * @param amountToLiquidate amount of liquidatedCollateral that liquidator wants to liquidate
+     * @param collateralPerToken abort the liquidation if the position's collateral per token exceeds this value
+     * @param maxTokensToLiquidate max number of tokes to liquidate
      * @return uuid of the newly created liquidation
      */
-    function createLiquidation(address sponsor, FixedPoint.Unsigned calldata amountToLiquidate)
-        external
-        fees()
-        onlyPreExpiration()
-        returns (uint uuid)
-    {
-        // Attempt to retrieve Position data for sponsor
+    // TODO: Perhaps pass this ID via an event rather than a return value
+    function createLiquidation(
+        address sponsor,
+        FixedPoint.Unsigned calldata collateralPerToken,
+        FixedPoint.Unsigned calldata maxTokensToLiquidate
+    ) external fees() onlyPreExpiration() returns (uint uuid) {
         PositionData storage positionToLiquidate = _getPositionData(sponsor);
-        FixedPoint.Unsigned memory positionCollateral = _getCollateral(positionToLiquidate.rawCollateral);
-        require(positionCollateral.isGreaterThan(0));
 
-        // Caller is required to include in order to prevent front-running attacks that modify the liquidatedCollateral amount
-        // @dev amountToLiquidate is >= underlying collateral because the position is still liquidatable if the underlying collateral declines
-        // while the liquidation transaction is pending
-        require(
-            amountToLiquidate.isGreaterThanOrEqual(positionCollateral.sub(positionToLiquidate.withdrawalRequestAmount))
+        // tokensToLiquidate = minimum of maxTokensToLiquidate and how many tokens are outstanding.
+        FixedPoint.Unsigned memory tokensToLiquidate = FixedPoint.Unsigned(maxTokensToLiquidate.rawValue).isGreaterThan(
+            positionToLiquidate.tokensOutstanding
+        )
+            ? positionToLiquidate.tokensOutstanding
+            : FixedPoint.Unsigned(maxTokensToLiquidate.rawValue);
+        // The ratio of the Position is getting liquidated, will be used to scale everything else.
+        FixedPoint.Unsigned memory ratio = tokensToLiquidate.div(positionToLiquidate.tokensOutstanding);
+
+        // Starting values for the Position being liquidated.
+        FixedPoint.Unsigned memory startCollateral = _getCollateral(positionToLiquidate.rawCollateral);
+        FixedPoint.Unsigned memory startCollateralNetOfWithdrawal = startCollateral.sub(
+            positionToLiquidate.withdrawalRequestAmount
         );
+        FixedPoint.Unsigned memory startTokens = positionToLiquidate.tokensOutstanding;
+
+        // Check that there is some collateral in the contract.
+        require(startCollateral.isGreaterThan(FixedPoint.Unsigned(0)));
```

Comment:
could this require not be replaced with the `onlyCollateralizedPosition` from the `PricelessPositionManager.sol` base contract?

Reply:
Yeah, and `onlyCollateralizedPosition` is already triggered on the call `_getPositionData` so agreed you don't need this line

Reply:
Removed it.